conf/pan/production_reg_conf.pl

```diff
@@ -35,35 +35,72 @@
+Bio::EnsEMBL::Registry->load_registry_from_url("mysql://ensro\@mysql-ens-sta-1:4519/$curr_release");
+# But remove the non-vertebrates species
+cleanup_adaptors_from_vertebrates_server('');
+# Non-Vertebrates server
 Bio::EnsEMBL::Registry->load_registry_from_url("mysql://ensro\@mysql-ens-sta-3:4160/$curr_release");
-#Bio::EnsEMBL::Registry->remove_DBAdaptor('saccharomyces_cerevisiae', 'core'); # never use EG's version of yeast
+# Bacteria server: all species used in Pan happen to be in this database
+Bio::EnsEMBL::Compara::Utils::Registry::load_collection_core_database(
+    -host   => 'mysql-ens-sta-4',
+    -port   => 4494,
+    -user   => 'ensro',
+    -pass   => '',
+    -dbname => "bacteria_0_collection_core_${curr_eg_release}_${curr_release}_1",
+);
 
 # ---------------------- PREVIOUS CORE DATABASES---------------------------------
 
 # previous release core databases will be required by PrepareMasterDatabaseForRelease and LoadMembers only
-# !!! COMMENT THIS SECTION OUT FOR ALL OTHER PIPELINES (for speed) !!!
-#my $suffix_separator = '__cut_here__';
-#Bio::EnsEMBL::Registry->load_registry_from_db(
-#    -host   => 'mysql-ens-mirror-3',
-#    -port   => 4275,
-#    -user   => 'ensro',
-#    -pass   => '',
-#    -db_version     => $prev_release,
-#    -species_suffix => $suffix_separator.$prev_release,
-#);
-#Bio::EnsEMBL::Registry->remove_DBAdaptor('saccharomyces_cerevisiae'.$suffix_separator.$prev_release, 'core'); # never use EG's version of yeast
-#Bio::EnsEMBL::Registry->load_registry_from_db(
-#    -host   => 'mysql-ens-mirror-1',
-#    -port   => 4240,
-#    -user   => 'ensro',
-#    -pass   => '',
-#    -db_version     => $prev_release,
-#    -species_suffix => $suffix_separator.$prev_release,
-#);
+*Bio::EnsEMBL::Compara::Utils::Registry::load_previous_core_databases = sub {
+    my $release_number  = shift;
+    my $species_suffix  = Bio::EnsEMBL::Compara::Utils::Registry::SUFFIX_SEPARATOR.$release_number;
+    Bio::EnsEMBL::Registry->load_registry_from_db(
+        -host   => 'mysql-ens-mirror-1',
+        -port   => 4240,
+        -user   => 'ensro',
+        -pass   => '',
+        -db_version     => $release_number,
+        -species_suffix => $species_suffix,
+    );
+    cleanup_adaptors_from_vertebrates_server("_${species_suffix}");
+    Bio::EnsEMBL::Registry->load_registry_from_db(
+        -host   => 'mysql-ens-mirror-3',
+        -port   => 4275,
+        -user   => 'ensro',
+        -pass   => '',
+        -db_version     => $release_number,
+        -species_suffix => $species_suffix,
+    );
+    # Bacteria server: all species used in Pan happen to be in this database
+    Bio::EnsEMBL::Compara::Utils::Registry::load_collection_core_database(
+        -host   => 'mysql-ens-mirror-4',
+        -port   => 4495,
+        -user   => 'ensro',
+        -pass   => '',
+        -dbname => "bacteria_0_collection_core_".($release_number-53)."_${release_number}_1",
+        -species_suffix => $species_suffix,
+    );
+};
```

Comment:
How to do this ? The list of the servers, their order, and the cleanup bit will change from one division to another

Reply:
You could make a function that accepts the list of mirrors and the `\@overlap_species`, and loads them in order, running the `remove_*` part between the first and the second mirrors (if more than 1 is given). So in the registry you would have something like the following in Pan:
```
Bio::EnsEMBL::Compara::Utils::Registry::load_previous_core_databases(['mysql-ens-mirror-1', 'mysql-ens-mirror-3', 'mysql-ens-mirror-4'], \@overlap_species);
```
and this for plants:
```
Bio::EnsEMBL::Compara::Utils::Registry::load_previous_core_databases(['mysql-ens-mirror-3', 'mysql-ens-mirror-1'], \@overlap_species);
```

Reply:
I'm not keen on having this specific operation inserted between the 1st and 2nd servers, but not between 2nd and 3rd, or 3rd and 4th, etc, with no control.
Thinking about this in the car, what about:
```perl
Bio::EnsEMBL::Compara::Utils::Registry::setup_previous_core_databases($prev_release,
    ['load_all_from_server', 'mysql-ens-mirror-3'],
    ['remove_species', \@overlap_species],
    ['remove_multi, undef],
    ['load_from_server', 'mysql-ens-mirror-1'],
);
```

By adding `load_all_from_server` to Utils::Registry and giving it the interface `($server_name, $release_number, $optional_species_suffix)`, `setup_previous_core_databases` could simply iterate over the array and for each element `$a` call the function named `$a->[0]` with the arguments `($a->[1], $prev_release, PREVIOUS_DATABASE_SUFFIX)`. For Pan we'd need a wrapper around `load_collection_core_database` with the same interface.

Then, with a slight adjustment, we can have `setup_current_core_databases` too, which I want to address in another PR anyway because it is touching more files and I don't want to make it too confusing.

Reply:
I agree with you, that would be the way to approach it if you want to have control on the steps within the method. But then I feel it starts to reduce the advantage of moving the function away from the registry, as "most" of the logic/code is still there...

Reply:
To me the advantage is that we move towards abstracting the definition of the Registry as a structure of hash / lists that could be moved into JSON and be read either from `production_reg_conf.pl` or whatever equivalent we will have in Python. Something like
```json
{
  "current_release": 100,
  "current_cores": [
    ["load_all_from_server", "mysql-ens-sta-3"],
    ["cleanup_overlap", ["saccharomyces_cerevisiae", "drosophila_melanogaster", "caenorhabditis_elegans"]],
    ["load_all_from_server", "mysql-ens-sta-1"],
  ],
  "previous_cores": [
    ["load_all_from_server", "mysql-ens-mirror-3"],
    ["cleanup_overlap", ["saccharomyces_cerevisiae", "drosophila_melanogaster", "caenorhabditis_elegans"]],
    ["load_all_from_server", "mysql-ens-mirror-1"],
  ],
  "aliases": {
    "compara": {
      "compara_master" : ["mysql-ens-compara-prod-5", "ensembl_compara_master_plants"],
      "compara_curr"   : ["mysql-ens-compara-prod-5", "ensembl_compara_plants_<curr_eg_release>_<curr_release>"],
      "compara_prev"   : ["mysql-ens-compara-prod-5", "ensembl_compara_plants_<prev_eg_release>_<prev_release>"],
    },
    "core": {
      "ancestral_curr" : ["mysql-ens-compara-prod-5", "ensembl_ancestral_plants_<curr_eg_release>_<curr_release>"],
    },
    "taxonomy": {
      "ncbi_taxonomy"  : ["mysql-ens-sta-1", "ncbi_taxonomy_<curr_release>" ],
    },
  },
}
```

Reply:
That's a really interesting approach! I quite like it :+1:

PS: I would add the `prev_release` key as an optional parameter just in case we want to use a different one than the default `-1`.